scipy/stats/stats.py

```diff
@@ -5718,3 +5721,195 @@ def rankdata(a, method='average'):
+    shift : float, optional.
+        Shift parameter of ratio-of-uniforms method, see Notes. Default is 0.
+    maxiter : int, optional
+        Maximum number of iterations to generate the random variables.
+        Default is None. In that case, maxiter is set to a value that
+        ensures successful simulation with a high probability.
+    bounds : tuple of floats (a, b), optional
+        Default is None. In case any of the values x1, x2, y2 are not
+        specified (i.e. equal to None), attempt to find the value numerically
+        over the interval from a to b.
+    random_state : None or int or np.random.RandomState instance, optional
+        If int or RandomState, use it for drawing the random variates.
+        If None, use np.random.RandomState. Default is None.
+
+    Returns
+    -------
+    rvs : array
+        the random variates distributed according to the probability
+        distrubtion, pdf
+
+    Notes
+    -----
+    Given a univariate probability density function (pdf) f and a constant c,
+    define the set A(c) = {(x, y) : 0 < y <= sqrt(f(x/y + c)) }.
+    If (U, V) is a random vector uniformly distributed over A(c), then U/V + c
+    follows a distribution with pdf f.
+
+    The above result (see [1]_, [2]_) can be used to sample random variables
+    using only the pdf, i.e. no inversion of the cdf is required. Typical
+    choices of c are zero or the mode of f. The set A(c) is a subset of the
+    rectangle R = [-x1, x2] x [0, y2] where
+    - x1 = inf (x - c) sqrt(f(x))
+    - x2 = sup (x - c) sqrt(f(x))
+    - y2 = sup sqrt(f(x))
+
+    In particular, these values are finite if f is bounded and x**2 * f(x) is
+    bounded (i.e. subquadratic tails). One can generate (U, V) uniformly on
+    R and return U/V + c if (U, V) are also in A(c) which can be directly
+    verified.
+
+    Intuitively, the method works well if A(c) fills up most of the
+    enclosing rectangle such that the probability is high that (U, V) lies in
+    A(c) whenever it lies in R(c) as the number of required iterations
+    becomes too large otherwise. To be more precise, note that the expected
+    number of iterations to draw (U, V) uniformly distributed on R(c) such that
+    (U, V) is also in A(c) is given by the ratio
+    area(R(c)) / area(A(c)) = 2 * y2 * (x2 - x1), using the fact that
+    the area of A(c) is equal to 1/2 (Theorem 7.1 in [1]_).
+
+    If the bounding rectangle is not correctly specified (i.e. if it does not
+    contain A(c)), the algorithm samples from a distribution different from
+    the one given by the pdf. It is therefore recommended to perform a
+    test such as `stats.kstest` as a check.
+
+    References
+    ----------
+    .. [1] L. Devroye, "Non-Uniform Random Variate Generation",
+       Springer-Verlag, 1986.
+
+    .. [2] W. Hoermann and J. Leydold, "Generating generalized inverse Gaussian
+       random variates", Statistics and Computing, 24(4), p. 547--557, 2014.
+
```

Comment:
Perhaps mention the 1977 paper which may be the earliest description of the algorithm. It also has useful examples giving the computation of regions for a few distributions.  BiBTex below
@article{Kinderman:1977:CGR:355744.355750,
 author = {Kinderman, A. J. and Monahan, J. F.},
 title = {Computer Generation of Random Variables Using the Ratio of Uniform Deviates},
 journal = {ACM Trans. Math. Softw.},
 issue_date = {Sept. 1977},
 volume = {3},
 number = {3},
 month = sep,
 year = {1977},
 issn = {0098-3500},
 pages = {257--260},
 numpages = {4},
 url = {http://doi.acm.org/10.1145/355744.355750},
 doi = {10.1145/355744.355750},
 acmid = {355750},
 publisher = {ACM},
 address = {New York, NY, USA},
}